File: js/findPeak.js

function findPeak(A) {
    let peak = 0;
    let peakIndex = 0;
    for (let i = 0; i < A.length; ++i) {
        if (A[i] > peak && A[i] > A[i-1] && A[i] > A[i+1]) {
            peak = A[i];
            peakIndex = i;
        }
    }
    return peakIndex;
}

console.log(findPeak([0, 2, 1, 0]));
console.log('hi');

File: py/findPeak.py

def findPeak(nums):
    peak = 0
    peakIndex = 0
    for i in range(len(nums)):
        if nums[i] > peak and nums[i] > nums[i-1] and nums[i] > nums[i+1]:
            peak = nums[i]
            peakIndex = i
    return peakIndex

print(findPeak([0, 2, 1, 0]))